tests/unit/src/test/scala/tests/DocumentHighlightSlowSuite.scala

```diff
@@ -0,0 +1,141 @@
+    for {
+      _ <- server.initialize(
+        s"""/metals.json
+           |{"a":{}}
+           |/$fileName
+           |object Main {
+           |  Option(1).head
+           |}
+        """.stripMargin
+      )
+      _ <- server.didOpen(fileName)
+      _ <- server.assertHighlight(
+        fileName,
+        """
+          |object Main {
+          |  Option(1).he@@ad
+          |}""".stripMargin,
+        """
+          |object Main {
+          |  Option(1).<<head>>
+          |}""".stripMargin
+      )
+    } yield ()
+  }
+
+  fileNamedTestAsync("multipleHighlight") { fileName =>
+    for {
+      _ <- server.initialize(
+        s"""/metals.json
+           |{"a":{}}
+           |/$fileName
+           |object Main {
+           |  val abc = 123
+           |  abc.toInt
+           |  println(abc)
+           |}
+        """.stripMargin
+      )
+      _ <- server.didOpen(fileName)
+      _ <- server.assertHighlight(
+        fileName,
+        """
+          |object Main {
+          |  val abc = 123
+          |  abc.toInt
+          |  println(ab@@c)
+          |}""".stripMargin,
+        """
+          |object Main {
+          |  val <<abc>> = 123
+          |  <<abc>>.toInt
+          |  println(<<abc>>)
+          |}""".stripMargin
+      )
+    } yield ()
+  }
+
+  fileNamedTestAsync("multipleDifferentSymbolsHighlight") { fileName =>
+    for {
+      _ <- server.initialize(
+        s"""/metals.json
+           |{"a":{}}
+           |/$fileName
+           |object Main {
+           |  val abc = 123
+           |  abc.toInt
+           |  134l.toInt
+           |}
+        """.stripMargin
+      )
+      _ <- server.didOpen(fileName)
+      _ <- server.assertHighlight(
+        fileName,
+        """
+          |object Main {
+          |  val abc = 123
+          |  abc.toInt@@
+          |  134l.toInt
+          |}""".stripMargin,
+        """
+          |object Main {
+          |  val abc = 123
+          |  abc.<<toInt>>
+          |  134l.toInt
+          |}""".stripMargin
+      )
+    } yield ()
+  }
+
+  fileNamedTestAsync("multipleHighlightWithEdit") { fileName =>
+    val additionalWhitespace = " " * 3
+    for {
+      _ <- server.initialize(
+        s"""/metals.json
+           |{"a":{}}
+           |/$fileName
+           |object Main {
+           |  val abc = 123
+           |  abc.toInt
+           |  println(abc)
+           |}
+        """.stripMargin
+      )
+      _ <- server.didOpen(fileName)
+      _ <- server.assertHighlight(
+        fileName,
+        s"""
+           |object Main {
+           |  val$additionalWhitespace abc = 123
+           |  abc.toInt
+           |  println(ab@@c)
+           |}""".stripMargin,
+        s"""
+           |object Main {
+           |  val$additionalWhitespace <<abc>> = 123
+           |  <<abc>>.toInt
+           |  println(<<abc>>)
+           |}""".stripMargin
+      )
+    } yield ()
+  }
+
+  private def fileNamedTestAsync(
```

Comment:
Alternatively, one pattern that I like a lot is to implement a `check` method like this

```scala
  def check(name: String, base: String, edit: String, expected: String): Unit =
    testAsync(name) {
      for {
        _ <- server.initialize(
          s"""/metals.json
             |{"a":{}}
             |/a/src/main/scala/a/Main.scala
             |$base
      """.stripMargin
        )
        _ <- server.didOpen("a/src/main/scala/a/Main.scala")
        _ <- server.assertHighlight(
          "a/src/main/scala/a/Main.scala",
          edit,
          expected
        )
      } yield ()
    }

  val additionalWhitespace = " " * 3
  check(
    "edit",
    s"""
       |object Main {
       |  val abc = 123
       |  abc.toInt
       |  println(abc)
       |}
        """.stripMargin,
    s"""
       |object Main {
       |  val$additionalWhitespace abc = 123
       |  abc.toInt
       |  println(ab@@c)
       |}""".stripMargin,
    s"""
       |object Main {
       |  val$additionalWhitespace <<abc>> = 123
       |  <<abc>>.toInt
       |  println(<<abc>>)
       |}""".stripMargin
  )
```

Ideally, ff we combine the original and the expected output then this can become as small as

```scala
check(
  "name",
   s"""|object Main {
       |  val <<abc>> = 123
       |  <<a@@bc>>.toInt
       |  println(<<abc>>)
       |}""".stripMargin
  )
)
```
We could automatically run the same test with an edit that prepends a newline to the file contents.

Reply:
I added the check method and a couple more cases.